### PR TITLE
feat: video block to use videos subgraph for video data

### PR DESCRIPTION
**Pull request by tataihono**
# Description

Please include a summary of the change and which todo is completed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- allow video to be included with variant filtering in other subgraphs
- refactor videoBlock to rely on video api for content (videoId and videoVariantLanguageId)
- refactor front end admin components to set videoId and videoVariantLanguageId fields

NB: This PR will need to be force merged as it introduces breaking API changes. Build failures are expected.

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] `nx seed api-videos` should run and add id to variant of video objects
- [ ] adding background video with Video ID of `2_0-FallingPlates` and Video Variant Language ID of `529` should work in journeys-admin editor
- [ ] adding video with Video ID of `2_0-FallingPlates` and Video Variant Language ID of `529` should work in journeys-admin editor

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-gateway/schema.graphql [hunk 1]
```diff
@@ -707,27 +707,22 @@ enum UserJourneyRole {
 
 type Video
   @join__owner(graph: VIDEOS)
-  @join__type(graph: VIDEOS, key: "id")
```
Comment (tataihono) [inline apps/api-gateway/schema.graphql:710]: primaryLanguageId can be null in this case.

File: apps/api-journeys/schema.graphql [hunk 2]
```diff
@@ -490,36 +490,19 @@ input TypographyBlockUpdateInput {
...
-type VideoGeneric implements VideoContent {
-  src: String
-}
-
 type VideoBlock implements Block {
   id: ID!
   journeyId: ID!
   parentBlockId: ID
   parentOrder: Int
-  title: String!
```
Comment (tataihono) [inline apps/api-journeys/schema.graphql:512]: defers to the video object to provide this.

File: apps/api-journeys/schema.graphql [hunk 3]
```diff
@@ -490,36 +490,19 @@ input TypographyBlockUpdateInput {
...
 
   """startAt dictates at which point of time the video should start playing"""
   startAt: Int
 
   """endAt dictates at which point of time the video should end"""
   endAt: Int
-  description: String
```
Comment (tataihono) [inline apps/api-journeys/schema.graphql:519]: defers to the video object to provide this.

File: apps/api-journeys/src/app/modules/block/video/video.resolver.ts [hunk 4]
```diff
@@ -49,22 +20,17 @@ export class VideoBlockResolver {
     ])
   )
   async videoBlockCreate(
-    @Args('input') input: VideoBlockCreateInput & { __typename }
+    @Args('input') input: VideoBlockCreateInput
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/video/video.resolver.ts:23]: input already enforces required ids

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Video/Video.spec.tsx [hunk 5]
```diff
@@ -28,6 +34,6 @@ describe('Video', () => {
     const { getByText } = render(<Video {...video} />)
 
     expect(getByText('Video Source')).toBeInTheDocument()
-    expect(getByText(video.title)).toBeInTheDocument()
+    expect(getByText(video.video?.variant?.hls ?? '')).toBeInTheDocument()
```
Comment (mikeallisonJS) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Video/Video.spec.tsx:37]: Any reason this has null checking on a const?

**Review by mikeallisonJS — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Video/Video.spec.tsx (hunk 5 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Video/Video.spec.tsx:37]: changed it to a static string

**Review by mikeallisonJS — APPROVED** [review]

**Pull request merged**